### PR TITLE
emacs: fix segfaults, add selinux capability

### PR DESCRIPTION
**Pull request by bendlas**
###### Motivation for this change

Fixes various segfaults in emacs 25.1
http://lists.gnu.org/archive/html/emacs-devel/2016-10/msg00917.html
https://debbugs.gnu.org/cgi/bugreport.cgi?bug=24358

The selinux addition is optional, but adds runtime capability to emacs.

###### Things done

- Built on platform(s)
   - [x] NixOS
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @ttuegel 

---

### CODE REVIEW

File: pkgs/applications/editors/emacs/default.nix [hunk 1]
```diff
@@ -34,13 +34,19 @@ stdenv.mkDerivation rec {
     sha256 = "0cwgyiyymnx4xdg99dm2drfxcyhy2jmyf0rkr9fwj9mwwf77kwhr";
   };
 
-  patches = lib.optional stdenv.isDarwin ./at-fdcwd.patch;
+  patches = (lib.optional stdenv.isDarwin ./at-fdcwd.patch) ++ [
```
Comment (ttuegel) [inline pkgs/applications/editors/emacs/default.nix]: If these patches are available from a stable URL, e.g. cgit, I think it's usually preferred to `fetchurl` them from there.

Comment (bendlas) [inline pkgs/applications/editors/emacs/default.nix]: Ok, I updated the commit.

**Comment by DamienCassou** [comment]
Should this be backported to 16.09?

**Comment by bendlas** [comment]
I've been wondering that. emacs 24 is default in 16.09, but for everybody running emacs 25, it should lead to a more stable, non-crashing experience.

**Comment by copumpkin** [comment]
This breaks darwin, which unsurprisingly does not support selinux 😄

**Comment by copumpkin** [comment]
@jwiegley fixed now

**Pull request merged**